runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java

```diff
@@ -315,7 +323,31 @@ public void onItemContainerChanged(final ItemContainerChanged event)
 			return;
 		}
 
-		inventoryItems = event.getItemContainer().getItems();
+		// Clone the array so changes aren't passed back to the event.
+		inventoryItems = event.getItemContainer().getItems().clone();
+
+		if (event.getItemContainer().contains(ItemID.RUNE_POUCH) || event.getItemContainer().contains(ItemID.RUNE_POUCH_L))
+		{
+			List<Item> runePouchContents = getRunepouchContents();
+
+			for (int i = 0; i < inventoryItems.length && runePouchContents.size() > 0; i++)
+			{
+				Item invItem = inventoryItems[i];
+				for (Item rune : runePouchContents)
+				{
+					if (invItem.getId() == rune.getId())
+					{
+						inventoryItems[i] = new Item(invItem.getId(), rune.getQuantity() + invItem.getQuantity());
```

Comment:
I guess this could overflow, but maybe it doesn't matter

Reply:
I could limit it if you think it's serious enough :man_shrugging: